userProfile/userProfile.mjs

```diff
@@ -0,0 +1,68 @@
+
+import DatabaseDriver from "../database/driver.mjs"
+
+// This module will use the TinyPEN API (RERUM Mongo DB)
+const database = new DatabaseDriver("mongo")
+
+export async function findUserById(id = null) {
+  try {
+    
+        let userProfile  = await database.find({ id: id })
+      // Needed to be getting from DB as above, but now as we dont have DB connection , just mocking as below
+      if(userProfile)
+      {
+        userProfile = {
+        id: id,
+        orchid_id: "0000-0000-3245-1188", // Dummy orchid ID
+        display_name: "Samply McSampleface" // Dummy display name
+        };
+      }
+      // adding special cases to test the error response
+      if(id === 111 || id === 222)
+      {
+        throw new Error("Internal Server Error");
```

Comment:
You need to actually respond with the error.  Right now, I caused the error, but the response code to the client was 200.  This is not functioning correctly.
![image](https://github.com/CenterForDigitalHumanities/TPEN-services/assets/3287006/b438a6e9-b802-465b-9651-bee78700f33e)

Reply:
updated to respond with error.